### PR TITLE
Report improvements

### PR DESCRIPTION
**Pull request by martingrossmann**
# Description

Added some features and fixes to the Testerra ReportNG

- Fixed Testerra version link to documentation (at the top right corner)
- Added run index to method details:
  image
- Added navigation between method details of tests with an error context (`Failed`, `Expected failed`, `Skipped`, tests with optionals) (#106)
  image
- Fixed search in `Logs` (#262)
- Added `@Fails` attributes `description` and `ticketString` and prompted messages to test view and made them searchable (#263, #277) 
  image
- Fixed color of repaired tests in thread view
- Updated the aurelia-components to latest version and migrated to intl-date-format-value-converter
- In some cases Selenium returns exceptions with embedded HTML. To prevent to interprete that messages an escape value converter was added. This change improves #255.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by martingrossmann** [comment]
@sbke-mms I've added a first version of navigation between failed methods.

File: report-ng/app/src/components/method-details/method.ts [hunk 1]
```diff
@@ -173,14 +176,34 @@ export class Method {
...
+        // Get all method details
+        this._statistics.getExecutionStatistics().then(executionStatistic => {
+            executionStatistic.classStatistics.forEach(classStatistic => {
+                classStatistic.methodContexts
+                    .filter(methodContext => methodContext.methodType == MethodType.TEST_METHOD)
+                    .forEach(methodContext => this._allMethodDetails.push(new MethodDetails(methodContext, classStatistic)));
+            });
+            // Sort for run index
+            this._allMethodDetails = this._allMethodDetails.sort((a, b) => a.methodContext.methodRunIndex - b.methodContext.methodRunIndex);
+        });
```
Comment (mreiche) [inline report-ng/app/src/components/method-details/method.ts:194]: Looks quite heavy to me to sort all methods, keep them cached and search them every time in the ViewModel to show the previous and next methods.

Please keep in mind, that functions cannot be observed in Aurelia. Thats why Aurelia is calling functions every 10ms. So your method `_getMethodDetailsFromCurrent()` gets called very often!

My lighweight proposal would be something like this:
```typescript
this._statistics.getExecutionStatistics().then(executionStatistics => {
    const myRunIndex = this._methodDetails.methodContext.methodRunIndex
    executionStatistics.classStatistics
        .flatMap(value => value.methodContexts)
        .filter(value => value.methodRunIndex === myRunIndex-1 || value.methodRunIndex === myRunIndex+1)
        .map(value => new MethodDetails(value, this._methodDetails.classStatistics))
        .forEach(value => {
            if (value.methodContext.methodRunIndex > myRunIndex) {
                this._nextMethod = value
            } else {
                this._prevMethod = value
            }
        })
})
```

```html
<div if.bind="_prevMethod">
    <span class="secondary sr1">Previous failed method</span>
     <a route-href="route: method; params.bind: {methodId: _prevMethod.methodContext.contextValues.id}" class="sr1">${_prevMethod.identifier}</a>
</div>
```

Comment (martingrossmann) [reply report-ng/app/src/components/method-details/method.ts:194]: Thanks for your hints. So best practice is to calculate complex stuff only via `activate` method and cache the results?

Comment (mreiche) [reply report-ng/app/src/components/method-details/method.ts:194]: For Aurelia:
- Prevent calling functions in the view, use observable properties instead.

General:
- Do it when needed
- Do it efficient
- Do it once

Comment (martingrossmann) [reply report-ng/app/src/components/method-details/method.ts:209]: I updated `methods.ts`. But I added some filters for test methods and failed tests. Finally based on a sort list to run index I can get the previous and the next method.

File: report-ng/app/src/components/method-details/method.ts [hunk 2]
```diff
@@ -173,14 +177,40 @@ export class Method {
...
+                * - filter for 'real' test methods (no setups)
+                * - sort for run index
+                * - create MethodDetails
+                * - filter for failed test methods
+                * - find the previous and the next failed test method to current.
+                 */
+                executionStatistics.classStatistics
+                    .flatMap(methodContext => methodContext.methodContexts)
+                    .filter(methodContext => methodContext.methodType == MethodType.TEST_METHOD)
+                    .sort((a, b) => a.methodRunIndex - b.methodRunIndex)
```
Comment (mreiche) [inline report-ng/app/src/components/method-details/method.ts:198]: Why sorting all methods when you need just 2? Does my approach not work here?

Comment (mreiche) [reply report-ng/app/src/components/method-details/method.ts:198]: Nevermind, I got it.

File: report-ng/app/src/components/method-details/method.html [hunk 3]
```diff
@@ -98,13 +99,38 @@
...
+                            <mdc-layout-grid-inner>
+                                <mdc-layout-grid-cell span="6">
+                                    <div if.bind="_prevMethod">
+                                        <span class="secondary sr1">Previous failed method</span>
+                                        <a route-href="route: method; params.bind: {methodId: _prevMethod.methodContext.contextValues.id}" class="sr1">${_prevMethod.identifier}</a>
+                                    </div>
+                                </mdc-layout-grid-cell>
+                                <mdc-layout-grid-cell span="6" class="right-aligned">
+                                    <div if.bind="_nextMethod">
+                                        <span class="secondary sl1 sr1"> Next failed method</span>
```
Comment (mreiche) [inline report-ng/app/src/components/method-details/method.html:120]: There is a suspect space left of "Next"

**Review by mreiche — APPROVED** [review]

File: report-ng/app/src/components/method-details/method.html (hunk 3 above)
Comment (martingrossmann) [reply report-ng/app/src/components/method-details/method.html]: Yes, that is not needed.

File: report-ng/app/src/services/status-converter.ts [hunk 4]
```diff
@@ -182,7 +182,7 @@ export class StatusConverter {
     }
 
     createRegexpFromSearchString(searchQuery:string) {
```
Comment (mreiche) [inline report-ng/app/src/services/status-converter.ts:184]: You can use https://www.npmjs.com/package/escape-string-regexp here

**Review by mreiche — APPROVED** [review]

**Review by mbeuthan — APPROVED** [review]

**Pull request merged**